Source/Scene/Expression.js

```diff
@@ -0,0 +1,1904 @@
+        } else if ((right instanceof Cartesian3) && (typeof left === 'number')) {
+            return Cartesian3.multiplyByScalar(right, left, ScratchStorage.getCartesian3());
+        } else if ((left instanceof Cartesian3) && (typeof right === 'number')) {
+            return Cartesian3.multiplyByScalar(left, right, ScratchStorage.getCartesian3());
+        } else if ((right instanceof Cartesian4) && (left instanceof Cartesian4)) {
+            return Cartesian4.multiplyComponents(left, right, ScratchStorage.getCartesian4());
+        } else if ((right instanceof Cartesian4) && (typeof left === 'number')) {
+            return Cartesian4.multiplyByScalar(right, left, ScratchStorage.getCartesian4());
+        } else if ((left instanceof Cartesian4) && (typeof right === 'number')) {
+            return Cartesian4.multiplyByScalar(left, right, ScratchStorage.getCartesian4());
+        } else if ((typeof left === 'number') && (typeof right === 'number')) {
+            return left * right;
+        }
+
+        //>>includeStart('debug', pragmas.debug);
+        throw new DeveloperError('Operator "*" requires vector or number arguments. If both arguments are vectors they must be matching types. Arguments are ' + left + ' and ' + right + '.');
+        //>>includeEnd('debug');
+
+        return left * right; // eslint-disable-line no-unreachable
+    };
+
+    Node.prototype._evaluateDivide = function(frameState, feature) {
+        var left = this._left.evaluate(frameState, feature);
+        var right = this._right.evaluate(frameState, feature);
+        if ((right instanceof Cartesian2) && (left instanceof Cartesian2)) {
+            return Cartesian2.divideComponents(left, right, ScratchStorage.getCartesian2());
+        } else if ((left instanceof Cartesian2) && (typeof right === 'number')) {
+            return Cartesian2.divideByScalar(left, right, ScratchStorage.getCartesian2());
+        } else if ((right instanceof Cartesian3) && (left instanceof Cartesian3)) {
+            return Cartesian3.divideComponents(left, right, ScratchStorage.getCartesian3());
+        } else if ((left instanceof Cartesian3) && (typeof right === 'number')) {
+            return Cartesian3.divideByScalar(left, right, ScratchStorage.getCartesian3());
+        } else if ((right instanceof Cartesian4) && (left instanceof Cartesian4)) {
+            return Cartesian4.divideComponents(left, right, ScratchStorage.getCartesian4());
+        } else if ((left instanceof Cartesian4) && (typeof right === 'number')) {
+            return Cartesian4.divideByScalar(left, right, ScratchStorage.getCartesian4());
+        } else if ((typeof left === 'number') && (typeof right === 'number')) {
+            return left / right;
+        }
+
+        //>>includeStart('debug', pragmas.debug);
+        throw new DeveloperError('Operator "/" requires vector or number arguments of matching types, or a number as the second argument. Arguments are ' + le
```

Comment:
Not sure how hot this switch is, but breaking all of these cases into smaller functions would probably help the browser optimize.